todo-app/src/Components/AddTodoItem.js

```diff
@@ -0,0 +1,99 @@
+//jshint esnext: true
+
+import React, { Component } from 'react'
+
+import Button from './Button';
+import InputField from './InputField';
+import './styles/addTodoItem.css';
+
+export default class AddTodoItem extends Component {
+
+    state = {
+        adding: false,
+        toBeAddedItem: '',
+        toBeAddedDeadline: ''
+    }
+
+
+
+    toggleAddFrom = event => {
+        this.setState({ adding: !this.state.adding });
+
+        //clearing fields for if it's a cancel
+        this.setState({
```

Comment:
Why do you have two `setState` calls here? You're not clearing the fields _only_ on a cancel, you're clearing them in both cases. So you should either combine your `setState` calls into a single call, or wrap the second call in a condition.

Also check the name of the function itself - small typo.